### PR TITLE
Feature/proj 100

### PR DESCRIPTION
**Pull request by EvgeniyaYakunina**
### **Description:**
This pull request introduces improvements to the authentication flow and the rendering of the UI components (Header and Sidebar) based on the user's authentication status.

### **Changes:**
**Token Storage and Retrieval:**

Implemented token storage and retrieval using localStorage via a Storage utility.
After successful login, the token is saved in localStorage and used in subsequent API requests.

### **Login Flow:**

Upon successful login, the token is extracted from the response, decoded to retrieve the userId, and the user is redirected to their profile page.
If the userId is not found in the token, a fallback request to authMe is made to get the user's data.
Base Query with Token Refresh Logic:

Added a baseQueryWithReauth function that handles token refreshing in case of 401 Unauthorized errors.
If a request fails due to an expired token, a new token is requested, saved, and the original request is retried.

### **Layout Component Logic:**

The Layout component checks the user's authentication status using useAuthMeQuery.
Displays the Header and Sidebar only when the user is authenticated (isSuccess).
Handles loading and error states:
Shows a "Loading..." message while authentication status is being checked.
If an error occurs, a message is displayed instead of redirecting the user.

### **Error Handling:**

If an error occurs during the authentication check, the layout will display an error message

### CODE REVIEW

File: src/features/auth/ui/signIn/useSignIn.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    },
+    mode: 'onBlur',
+    resolver: zodResolver(signInSchema),
+  })
+  const [signIn] = useSignInMutation()
+  const [getMe] = useLazyAuthMeQuery()
+  const onSubmit = handleSubmit(data => {
+    signIn(data)
+      .unwrap()
+      .then(async res => {
```
Comment (Lelik7777) [inline src/features/auth/ui/signIn/useSignIn.tsx:43]: стоит добавить js doc comment для обьяснения

File: src/app/store/store.ts [hunk 2]
```diff
@@ -1,15 +1,14 @@
...
-export const store = configureStore({
-  middleware: getDefaultMiddleware => getDefaultMiddleware().concat(baseApi.middleware),
-  reducer: {
-    [baseApi.reducerPath]: baseApi.reducer,
-  },
-})
+const makeStore = () =>
+  configureStore({
+    middleware: getDefaultMiddleware => getDefaultMiddleware().concat(baseApi.middleware),
+    reducer: combineSlices(baseApi),
```
Comment (Lelik7777) [inline src/app/store/store.ts:8]: стоит добавить js doc comment для обьяснения

File: src/features/layout/layout.tsx [hunk 3]
```diff
@@ -21,26 +24,25 @@ import style from './layout.module.scss'
...
  */
 
-// eslint-disable-next-line react-refresh/only-export-components
-const Layout = ({ children }: PropsWithChildren) => {
+export const Layout = ({ children }: { children: ReactNode }) => {
+  const router = useRouter()
+  const { isError, isLoading, isSuccess } = useAuthMeQuery()
+
+  useEffect(() => {
+    if (isError) {
```
Comment (Lelik7777) [inline src/features/layout/layout.tsx:32]: TODO: add logic for handle error

File: src/features/layout/layout.tsx [hunk 4]
```diff
@@ -21,26 +24,25 @@ import style from './layout.module.scss'
...
+      console.error(isError)
+    }
+  }, [isError, router])
+
+  if (isLoading) {
+    return <div>Loading...</div>
```
Comment (Lelik7777) [inline src/features/layout/layout.tsx:38]: TODO: add logic for show loarding

File: src/pages/profile/[userId].tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { useAuthMeQuery } from '@/features'
+import { useRouter } from 'next/router'
+
+const ProfilePage = () => {
+  const router = useRouter()
+  const { data, isSuccess } = useAuthMeQuery()
+  const userId = router.query.userId
+
+  if (!isSuccess) {
```
Comment (Lelik7777) [inline src/pages/profile/[userId].tsx:9]: TODO: add logic for show loading

File: src/features/auth/ui/signIn/useSignIn.tsx (hunk 1 above)
Comment (EvgeniyaYakunina) [reply src/features/auth/ui/signIn/useSignIn.tsx:43]: добавила

File: src/app/store/store.ts (hunk 2 above)
Comment (EvgeniyaYakunina) [reply src/app/store/store.ts:8]: добавила

**Pull request merged**